robot/basestation/static/js/elements/feed/camera.js

```diff
@@ -14,204 +14,321 @@ $(document).ready(() => {
-      requestTask(
-        CAMERA_TASK,
-        0,
-        '#front-camera-stream-btn',
-        function (msgs) {
-          printErrToConsole(msgs)
-          console.log('front camera OFF msgs:', msgs)
-          if (msgs[1].includes(CAMERA_STOP_SUCCESS_RESPONSE)) {
-            // succeeded to close stream
-            $('img#camera-feed')[0].src = STREAM_OFF
-            $('img#camera-feed').removeClass('rotateimg180')
-          } else {
-            appendToConsole('Failed to close stream')
-          }
-        },
-        FRONT_PORT
-      )
-    }
-  })
+  const STREAM_OFF = '../../static/img/camera/stream-offline.png'
+  const POWER_ON = '../../static/img/camera/power_on.png';
+  const POWER_OFF = '../../static/img/camera/power_off.png';
 
-  $('#rear-camera-stream-btn').on('click', function (event) {
-    event.preventDefault()
-    // click makes it checked during this time, so trying to enable
-    if ($('#rear-camera-stream-btn').is(':checked')) {
-      requestTask(
-        CAMERA_TASK,
-        1,
-        '#rear-camera-stream-btn',
-        function (msgs) {
-          printErrToConsole(msgs)
-          console.log('rear camera ON msgs:', msgs)
-          if (msgs[1].includes(CAMERA_START_SUCCESS_RESPONSE)) {
-            $('img#camera-feed')[0].src = getStreamURL(REAR_STREAM_TOPIC)
-            console.log('new src: ' + $('img#camera-feed')[0].src)
-          } else {
-            appendToConsole('Failed to open stream')
-            $('#rear-camera-stream-btn')[0].checked = false
-          }
-        },
-        REAR_PORT
-      )
-    } else {
-      requestTask(
-        CAMERA_TASK,
-        0,
-        '#rear-camera-stream-btn',
-        function (msgs) {
-          printErrToConsole(msgs)
-          if (msgs[1].includes(CAMERA_STOP_SUCCESS_RESPONSE)) {
-            console.log('rear camera OFF msgs:', msgs)
-
-            // succeeded to close stream
-            $('img#camera-feed')[0].src = STREAM_OFF
-          } else {
-            // failed to close stream
-            appendToConsole('Failed to close stream')
-          }
-        },
-        REAR_PORT
+  const RECORDING_ON = "../../../static/img/camera/record_on.png";
+  const RECORDING_OFF = "../../../static/img/camera/record_off.png";
+
+  function getStreamURL(topicName) {
+    return 'http://' + getRoverIP() + ':8080/stream?topic=/' + topicName + '/image_raw'
+  }
+
+  function getCameraFilename(cameraPanel) {
+    let cameraName = getCameraName(cameraPanel);
+    let cameraNameSplit = cameraName.split('/');
+    let cameraFilename = cameraNameSplit[cameraNameSplit.length - 1]
+    return cameraFilename;
+  }
+
+  function getCameraName(cameraPanel){
+    let cameraNameElement = cameraPanel.find(".camera-name");
+    let cameraName = cameraNameElement.attr("stream");
+    return cameraName;
+  }
+
+  function startCameraStream(cameraStream, successCallback = () => {}) {
+    requestTask(CAMERA_STREAM, STATUS_START, msgs => {
+        if(msgs[0])
```

Comment:
CS --> always use `{}`, even for single body line if/else blocks
CS --> semicolons in JS? :angry:

Reply:
done